clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -204,6 +252,28 @@ class AzureNetworkClient extends AzureBaseClient {
     result
   }
 
+  /**
+   * Retrieve an Azure network security group for a give set of credentials, resource group and network security group name
+   * @param creds the credentials to use when communicating to the Azure subscription(s)
+   * @param resourceGroupName name of the resource group where the security group was created (see application name and region/location)
+   * @param securityGroupName name of the Azure network security group to be retrieved
+   * @return a description object which represent a Network Security Group in Azure
+   */
+  AzureSecurityGroupDescription getNetworkSecurityGroup(AzureCredentials creds, String resourceGroupName, String securityGroupName) {
+    try {
+      def securityGroup = this.getNetworkResourceProviderClient(creds).getNetworkSecurityGroupsOperations().get(resourceGroupName, securityGroupName).networkSecurityGroup
+      def currentTime = System.currentTimeMillis()
+      def sgItem = getAzureSecurityGroupDescription(securityGroup)
+      sgItem.lastReadTime = currentTime
+
+      return sgItem
+    } catch (Exception e) {
+      log.info("getNetworkSecurityGroupsAll -> Unexpected exception " + e.toString())
```

Comment:
Yep, makes sense to use log.error() in this case; I'll make the necessary changes.